### PR TITLE
Encrypted volumes

### PR DESCRIPTION
**Pull request by AdrienClairembault**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   |

### CODE REVIEW

**Comment by cedric-anne** [comment]
You have to update your search options identifier. Tests results: `Duplicate key 157 (Encryption status/Wifi network) in CommonDBTM searchOptionsToAdd for Computer!`.

File: inc/item_disk.class.php [hunk 1]
```diff
@@ -337,6 +342,27 @@ static function showForItem(CommonDBTM $item, $withtemplate = 0) {
...
+                  Dropdown::getYesNo($data['encryption_status'] == self::ENCRYPTION_STATUS_PARTIALLY) .
+                  "<br/>" .
+                  "<strong>" . __('Encryption tool') . "</strong> : " . $data['encryption_tool'] .
+                  "</br>" .
+                  "<strong>" . __('Encryption algorithm') . "</strong> : " .
+                  $data['encryption_algorithm'] . "</br>" .
+                  "<strong>" . __('Encryption type') . "</strong> : " . $data['encryption_type'] .
+                  "</br>";
+
+               echo "<i id=\"$encryptionID\" class=\"fas fa-lock\"></i> &nbsp;";
```
Comment (cedric-anne) [inline inc/item_disk.class.php]: You can use the `showTooltip` method options to simplify your code:
```
               Html::showTooltip($encryptionTooltip, [
                  'awesome-class' => "fas fa-lock",
               ]);
```

By doiing this, you can remove echo of the picture and remove the definition of `$encryptionID` var.

File: inc/item_disk.class.php [hunk 2]
```diff
@@ -337,6 +342,27 @@ static function showForItem(CommonDBTM $item, $withtemplate = 0) {
...
+               echo Html::showTooltip($encryptionTooltip, [
```
Comment (cedric-anne) [inline inc/item_disk.class.php]: `echo` should be removed `showTooltip` does already it by default.
```suggestion
               Html::showTooltip($encryptionTooltip, [
```

**Comment by cedric-anne** [comment]
OK for me, but migrations must be put in `update_94_100.php` prior to merge.

File: inc/item_disk.class.php [hunk 3]
```diff
@@ -464,6 +506,121 @@ public static function rawSearchOptionsToAdd($itemtype) {
...
+         'datatype'           => 'string',
+         'joinparams'         => [
+            'jointype'           => 'itemtype_item'
+         ]
+      ];
+
       return $tab;
    }
+
+   static function getEncryptionStatusDropdown($value = self::ENCRYPTION_STATUS_NO, $options = []) {
```
Comment (trasher) [inline inc/item_disk.class.php]: `get` is not OK, this we do not return anything. Please also add phpdoc

Comment (trasher) [reply inc/item_disk.class.php]: BTW, note that new UI will certainly require at some point to get possible status list (`$allStatus`) from a real `getSthing` ;)

File: inc/item_disk.class.php [hunk 4]
```diff
@@ -464,6 +506,121 @@ public static function rawSearchOptionsToAdd($itemtype) {
...
+
+      switch ($field) {
+         case 'encryption_status':
+            return self::getEncryptionStatus($values[$field]);
+      }
+
+      return parent::getSpecificValueToDisplay($field, $values, $options);
+   }
+
+   static function getEncryptionStatus($status) {
```
Comment (trasher) [inline inc/item_disk.class.php]: Missing phpdoc

**Comment by trasher** [comment]
For the new UI part, I had to make some adaptations in the form builder; I'll take care of that in another pull request.

**Pull request merged**